### PR TITLE
Shortform flags

### PR DESCRIPTION
**Pull request by jimrandomh**
Adds a new bit of admin-only UI to PostActions to make a post be the short-form feed for a user. Adds a shortform flag on posts and comments, and callbacks to keep it up to date.

(This will be followed up with UI that makes a shortform page, puts shortform stuff on All Posts, and a front-page comment-box for short-form writing which makes you a shortform post if you don't already have one.)

### CODE REVIEW

**Comment by Raemon** [comment]
Tests to run (not run yet)
 – Set as a user's shortform updates the user object and post object
 – Move to frontpage and setAlignmentPost still work
 – Manually setting a user's shortformFeedID updates the post in question

**Comment by jimrandomh** [comment]
>  – Set as a user's shortform updates the user object and post object

Tested

> – Move to frontpage and setAlignmentPost still work

These don't seem at all relevant?

> – Manually setting a user's shortformFeedID updates the post in question

I consider that to be bypassing the API in an unsupported way. If you use admin access to muck with internal fields, when there's a perfectly good UI for updating those fields, you'll break things.

File: packages/lesswrong/lib/collections/posts/callbacks.js [hunk 1]
```diff
@@ -136,5 +137,18 @@ function PostsNewPostRelation (post) {
...
 addCallback("posts.new.after", PostsNewPostRelation);
+
+function UpdatePostShortform (newPost, oldPost) {
+  if (!!newPost.shortform !== !!oldPost.shortform) {
+    const shortform = !!newPost.shortform;
+    Comments.update(
+      { postId: newPost._id },
+      { $set: {
+        shortform: shortform
+      } }
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/callbacks.js]: I am pretty sure there is a `multi: true` missing here

File: packages/lesswrong/lib/collections/posts/schema.js [hunk 2]
```diff
@@ -447,7 +447,20 @@ const schema = {
   'targetPostRelations.$': {
     type: String,
     optional: true,
-  }
+  },
+  
+  // A post should have the shortform flag set iff its author's shortformFeedId
+  // field is set to this post's ID.
+  shortform: {
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/schema.js]: Shouldn't this then utilize the `denormalizedField` abstraction?

File: packages/lesswrong/lib/collections/users/callbacks.js [hunk 3]
```diff
@@ -139,3 +139,35 @@ async function subscribeOnSignup (user) {
...
+    if (!post)
+      throw new Error("Invalid post ID for shortform");
+    if (post.userId !== newUser._id)
+      throw new Error("Post can only be an author's short-form post if they are the post's author");
+    if (post.draft)
+      throw new Error("Draft post cannot be a user's short-form post");
+    if (post.deleted)
+      throw new Error("Deleted post cannot be a user's short-form post");
+    
+    // In theory, we should check here whether the user already had a short-form
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/users/callbacks.js]: I expect we will have to add this functionality eventually, but it's fine for now

**Comment by Discordius** [comment]
I expect this to currently only move a single comment on a post to have `shortform: true` instead of all of them, due to the missing `multi: true` in the update call, but not confident

**Pull request merged**